### PR TITLE
[refactor] CLI changes

### PR DESCRIPTION
**Pull request by ervteng**
### Proposed change(s)

- Make `--train` the default, and deprecate it (with warning message)
- Add `--inference` flag for inference mode
- Replace `--load` with `--resume`
- Block training if the model files already exist. Require `--force` to overwrite these files.
- Tensorboard writer deletes old summaries if overwriting. This prevents the double-back-around plots we sometimes get if we use the same run-id.

~TODO: Doc changes, changelog~

### Useful links (Github issues, JIRA tickets, ML-Agents forum threads etc.)
MLA-787

### Types of change(s)

- [ ] Bug fix
- [ ] New feature
- [x] Code refactor
- [ ] Breaking change
- [ ] Documentation update
- [ ] Other (please describe)

### Checklist
- [x] Added tests that prove my fix is effective or that my feature works
- [ ] Updated the [changelog](https://github.com/Unity-Technologies/ml-agents/blob/master/com.unity.ml-agents/CHANGELOG.md) (if applicable)
- [ ] Updated the [documentation](https://github.com/Unity-Technologies/ml-agents/tree/master/docs) (if applicable)
- [ ] Updated the [migration guide](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Migrating.md) (if applicable)

### CODE REVIEW

File: ml-agents/mlagents/trainers/trainer_util.py [hunk 1]
```diff
@@ -190,3 +190,33 @@ def _load_config(fp: TextIO) -> Dict[str, Any]:
             "Error parsing yaml file. Please check for formatting errors. "
             "A tool such as http://www.yamllint.com/ can be helpful with this."
         ) from e
+
+
+def handle_existing_directories(
+    model_path: str, summary_path: str, resume: bool, force: bool
```
Comment (ervteng) [inline ml-agents/mlagents/trainers/trainer_util.py]: note that `summary_path` isn't used yet but with consolidating all summaries into the same directory will be checked as well.

File: ml-agents/mlagents/trainers/stats.py [hunk 2]
```diff
@@ -198,8 +198,21 @@ def _maybe_create_summary_writer(self, category: str) -> None:
                 basedir=self.base_dir, category=category
             )
             os.makedirs(filewriter_dir, exist_ok=True)
+            self._delete_all_events_files(filewriter_dir)
```
Comment (harperj) [inline ml-agents/mlagents/trainers/stats.py]: Wouldn't this be a problem when using `--resume`?

Comment (ervteng) [reply ml-agents/mlagents/trainers/stats.py]: Added a fix - the TB writer now takes in a param that tells it to not delete if `--resume` is passed.

File: ml-agents/mlagents/trainers/stats.py [hunk 3]
```diff
@@ -199,8 +202,22 @@ def _maybe_create_summary_writer(self, category: str) -> None:
                 basedir=self.base_dir, category=category
             )
             os.makedirs(filewriter_dir, exist_ok=True)
+            if self._clear_past_data:
+                self._delete_all_events_files(filewriter_dir)
             self.summary_writers[category] = tf.summary.FileWriter(filewriter_dir)
 
+    def _delete_all_events_files(self, directory_name: str) -> None:
```
Comment (harperj) [inline ml-agents/mlagents/trainers/stats.py]: Minor: could add some logging to make the user aware this is happening.

Comment (ervteng) [reply ml-agents/mlagents/trainers/stats.py]: Added

**Pull request merged**